src/Workspaces/Remote/Core/SolutionAssetProvider.cs

```diff
@@ -3,230 +3,63 @@
-
-        private readonly SolutionServices _services = services;
-
-        public ValueTask WriteAssetsAsync(
-            PipeWriter pipeWriter,
-            Checksum solutionChecksum,
-            AssetPath assetPath,
-            ReadOnlyMemory<Checksum> checksums,
-            CancellationToken cancellationToken)
-        {
-            // Suppress ExecutionContext flow for asynchronous operations operate on the pipe. In addition to avoiding
-            // ExecutionContext allocations, this clears the LogicalCallContext and avoids the need to clone data set by
-            // CallContext.LogicalSetData at each yielding await in the task tree.
-            //
-            // ⚠ DO NOT AWAIT INSIDE THE USING. The Dispose method that restores ExecutionContext flow must run on the
-            // same thread where SuppressFlow was originally run.
-            using var _ = FlowControlHelper.TrySuppressFlow();
-            return WriteAssetsSuppressedFlowAsync(pipeWriter, solutionChecksum, assetPath, checksums, cancellationToken);
-
-            async ValueTask WriteAssetsSuppressedFlowAsync(PipeWriter pipeWriter, Checksum solutionChecksum, AssetPath assetPath, ReadOnlyMemory<Checksum> checksums, CancellationToken cancellationToken)
-            {
-                // The responsibility is on us (as per the requirements of RemoteCallback.InvokeAsync) to Complete the
-                // pipewriter.  This will signal to streamjsonrpc that the writer passed into it is complete, which will
-                // allow the calling side know to stop reading results.
-                Exception? exception = null;
-                try
-                {
-                    await WriteAssetsWorkerAsync(pipeWriter, solutionChecksum, assetPath, checksums, cancellationToken).ConfigureAwait(false);
-                }
-                catch (Exception ex) when ((exception = ex) == null)
-                {
-                    throw ExceptionUtilities.Unreachable();
-                }
-                finally
-                {
-                    await pipeWriter.CompleteAsync(exception).ConfigureAwait(false);
-                }
-            }
-        }
-
-        private async ValueTask WriteAssetsWorkerAsync(
-            PipeWriter pipeWriter,
-            Checksum solutionChecksum,
-            AssetPath assetPath,
-            ReadOnlyMemory<Checksum> checksums,
-            CancellationToken cancellationToken)
-        {
-            var assetStorage = _services.GetRequiredService<ISolutionAssetStorageProvider>().AssetStorage;
-            var serializer = _services.GetRequiredService<ISerializerService>();
-            var scope = assetStorage.GetScope(solutionChecksum);
+namespace Microsoft.CodeAnalysis.Remote;
 
-            using var _ = Creator.CreateResultMap(out var resultMap);
-
-            await scope.AddAssetsAsync(assetPath, checksums, resultMap, cancellationToken).ConfigureAwait(false);
```

Comment:
all this logic moved into WriteDataAsync.  so this method was inlined into the callsite and became practically nothing.